src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,113 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final String READ_ERROR_MSG = "Can't read from this file: ";
+    private static final String WRITE_ERROR_MSG = "Can't write to this file: ";
+    private static final String SEPARATOR_REGEX = ",";
+    private static final String RESULT_LINE = "result,%s";
+    private static final String SUPPLY_ALWAYS_FIRST = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFile(String fromFileName) {
+        StringBuilder dataFromFile = new StringBuilder();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+
+            while (value != null) {
+                String[] parts = value.split(SEPARATOR_REGEX);
+                for (String part : parts) {
+                    dataFromFile.append(part).append(SEPARATOR_REGEX);
+                }
+                value = reader.readLine();
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException(READ_ERROR_MSG + fromFileName + e);
```

Comment:
Pay attention - we should pass 2 arguments 
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/08c9c44a-3274-4099-a5b7-a7b9cfe5f8c9)